server/index.js

```diff
@@ -62,6 +62,106 @@ app.get('/api/products/:productId', (req, res, next) => {
 }
 );
 
+app.get('/api/cart', (req, res, next) => {
+  if (!req.session.cartId) {
+    return res.status(200).json([]);
+  } else {
+    const sqlSelect = `
+  select "c"."cartItemId",
+        "c"."price",
+        "p"."productId",
+        "p"."image",
+        "p"."name",
+        "p"."shortDescription"
+    from "cartItems" as "c"
+    join "products" as "p" using ("productId")
+    where "c"."cartId" = $1
+    `;
+    const queryValue = [req.session.cartId];
+    db.query(sqlSelect, queryValue)
+      .then(result => {
+        res.json(result.rows);
+      })
+      .catch(err => next(err));
+  }
+});
+
+app.post('/api/cart', (req, res, next) => {
+  if (isNaN(req.body.productId)) {
+    return res.status(400).json({ error: `Invalid field used for this POST method for ProductId '${req.body.productId}'. Please correct property syntax or try using a number type value.` });
+  }
+  if (req.body.productId < 0) {
+    return res.status(400).json({ error: `Invalid field used for this POST method for ProductId '${req.body.productId}'. Please use an Id greater than 0.` });
+  }
+  const sqlSelect = `
+  select
+  "price",
+  "productId"
+  from
+    "products"
+  where "productId" = $1
+  `;
+  const values = [req.body.productId];
+  db.query(sqlSelect, values)
+    .then(result => {
+      if (result.rows.length === 0) {
+        throw next(new ClientError(`Product Id: ${req.body.productId} cannot be found`, 400));
+      }
+      const priceAndProductId = result.rows[0];
+      if (req.session.cartId !== undefined) {
+        const cartIdSessionObj = { cartId: req.session.cartId };
+        const returnObj = Object.assign(cartIdSessionObj, priceAndProductId);
+        return returnObj;
+      } else {
+        const sqlInsert = `
+        insert into "carts" ("cartId", "createdAt")
+        values (default, default)
+        returning "cartId";`;
+        return db.query(sqlInsert)
+          .then(result => {
+            const returnObj = Object.assign(result.rows[0], priceAndProductId);
+            return returnObj;
+          })
+          .catch(err => next(err));
+      }
+    })
+    .then(result => {
+      const returnedCartId = result.cartId;
+      req.session.cartId = returnedCartId;
+      const sqlInsert = `
+      insert into "cartItems" ("cartId", "productId", "price")
+      values ($1, $2, $3)
+      returning "cartItemId"
+      `;
+      const values = [result.cartId, result.productId, result.price];
+      return db.query(sqlInsert, values)
+        .then(result => {
+          return result.rows[0];
+        })
+        .catch(err => next(err));
```

Comment:
You should remove this `.catch()` and let your final `.catch()` handle the promise rejection instead.